.github/workflows/docker.yml

```diff
@@ -64,3 +80,37 @@ jobs:
           pull: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Sign the published Docker image with GitHub OIDC Token
+        env: 
+          DIGEST: ${{ steps.build-and-push-docker-image.outputs.digest }}
+          TAGS: ${{ steps.meta.outputs.tags }}
+        run: |
+          images=""
+          for tag in ${TAGS}; do
+            images+="${tag}@${DIGEST} "
+          done
+          cosign sign --yes ${images}
+        
+      - name: Output image
+        id: image
+        run: |
+          # NOTE: Set the image as an output because the `env` context is not
+          # available to the inputs of a reusable workflow call.
+          image_name="${IMAGE_REGISTRY}/${IMAGE_NAME}"
+          echo "image=$image_name" >> "$GITHUB_OUTPUT"
+        
+  provenance:
+    needs: [build-and-push-image]
+    permissions:
+      actions: read # for detecting the Github Actions environment.
+      id-token: write # for creating OIDC tokens for signing.
+      packages: write # for uploading attestations.
+    if: github.repository == 'restic/restic'
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v2.0.0
```

Comment:
Is there a downside to using the generator framework? I'd expect it to be less work than using `cosign` directly

Reply:
Yes, it's lesser work than the container signing. I will be creating a separate Issue for that. 

In terms of security I reckon both are required, that's why I raised it in a single PR. 

Container Signing would be required for integrity and SLSA Provenance would be required for the origins of the build process.